File: VNC.CodeAnalysis Test Cases/TestCasesVB/Quality/MagicNumbersInIndex.vb
Public Class MagicNumbersInIndex
    Public Function fun() As String
        Dim dic As Dictionary(Of Integer, String) = New Dictionary(Of Integer, String)()
        Dim x As Integer = 323
        Return dic(x) + x + dic(323)
    End Function

    Public Function funny() As Single
        Dim d As Integer = 234
        Dim dic As Dictionary(Of Integer, Single) = New Dictionary(Of Integer, Single)()
        Return dic(d)
    End Function
End Class

File: CodeAnalysis Test Cases/TestCasesCS/Quality/MagicNumbersInIndex.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace CSTestCases.Quality
{
    public class MagicNumbersInIndex
    {
        public string fun()
        {
            Dictionary<int, string> dic = new Dictionary<int, string>();

            int x = 323;
            string z = dic[x] + x + dic[323];

            return z;
        }

        public float funny()
        {
            int d = 234;
            Dictionary<int, float> dic = new Dictionary<int, float>();
            float z = dic[d];
            return z;
        }
    }
}
